src/Microsoft.Extensions.Http/HttpClientBuilder.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Microsoft.Extensions.Http
+{
+    /// <summary>
+    /// A builder abstraction for configuring <see cref="HttpMessageHandler"/> instances.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="HttpMessageHandlerBuilder"/> is registered in the service collection as
+    /// a transient service. Callers should retrieve a new instance for each <see cref="HttpMessageHandler"/> to
+    /// be created. Implementors should expect each instance to be used a single time.
+    /// </remarks>
+    public abstract class HttpMessageHandlerBuilder
```

Comment:
This is a replaceable abstraction. You can act as a 'coordinator' for the handlers, or configure your own set of default handlers.

Reply:
File and class names are different.

Reply:
Hmm, there's something similar in HttpAbstractions with the `IAppliactonBuilder` and the `IApplicaitonBuilderFactory`.

Reply:
Yeah I think it's a pivot here that we could use DI as the factory vs create a factory abstraction. The thing is that we don't pass any parameters to the handler builder and I don't anticipate that we will - it's already surrounded by a composition mechanism.

I think using DI for a factory is the simple/right approach. If you think this is too daring let's talk more.

Reply:
> File and class names are different.

Thanks, will fix.